arch/arm/src/common/Make.defs

```diff
@@ -23,12 +23,18 @@
 CMN_CSRCS += arm_allocateheap.c arm_assert.c arm_blocktask.c
 CMN_CSRCS += arm_createstack.c arm_exit.c arm_fullcontextrestore.c
 CMN_CSRCS += arm_initialize.c arm_interruptcontext.c arm_lowputs.c
-CMN_CSRCS += arm_mdelay.c arm_modifyreg16.c arm_modifyreg32.c
+CMN_CSRCS += arm_modifyreg16.c arm_modifyreg32.c
 CMN_CSRCS += arm_modifyreg8.c arm_puts.c arm_releasepending.c
 CMN_CSRCS += arm_releasestack.c arm_reprioritizertr.c arm_saveusercontext.c
-CMN_CSRCS += arm_stackframe.c arm_switchcontext.c arm_udelay.c
+CMN_CSRCS += arm_stackframe.c arm_switchcontext.c
 CMN_CSRCS += arm_vfork.c arm_unblocktask.c arm_usestack.c
 
+ifneq ($(CONFIG_ALARM_ARCH),y)
+  ifneq ($(CONFIG_TIMER_ARCH),y)
```

Comment:
could you please provide some more explanation why we have this dependency?

Reply:
Since arch alarm/timer implement up_mdelay/up_udelay too:
https://github.com/apache/incubator-nuttx/blob/master/drivers/timers/arch_alarm.c#L401-L435
https://github.com/apache/incubator-nuttx/blob/master/drivers/timers/arch_timer.c#L459-L484
Both implementation is more general and less configuration than common's, so the change prefer to the first implementation.

Reply:
I see. Thank you